.evergreen/config.yml

```diff
@@ -1,48 +1,191 @@
+# Mark a failure as a system/bootstrap failure (purple box) rather then a task
+# failure by default.
+# Actual testing tasks are marked with `type: test`
+command_type: system
+
+# Protect ourselves against rogue test case, or curl gone wild, that runs forever
+exec_timeout_secs: 3600
+
+# When a task that used to pass starts to fail, go through all versions that may have been skipped
+# to detect when the task started failing
+stepback: true
+
+#######################################
+#              Functions              #
+#######################################
 functions:
-  fetch source:
-    # Executes git clone
+  "fetch source":
+    # Executes git clone and applies the submitted patch, if any.
     - command: git.get_project
       params:
         directory: src
-    # Applies the submitted patch, if any
-    - command: git.apply_patch
 
-  exec script:
-    - command: shell.exec
-      type: test
+  "create expansions":
+    - command: subprocess.exec
+      params:
+        working_dir: src
+        binary: bash
+        args:
+          - .evergreen/create-expansions.sh
+        include_expansions_in_env:
+          - is_patch
+          - version_id
+          - project
+    - command: expansions.update
+      params:
+        file: src/expansion.yml
+
+  "prepare resources":
+    - command: subprocess.exec
       params:
         working_dir: src
-        shell: bash
-        add_to_path:
-          - .evergreen
+        binary: bash
+        args:
+          - .evergreen/fetch-drivers-tools.sh
+        include_expansions_in_env:
+          - DRIVERS_TOOLS
+    - command: shell.exec
```

Comment:
This is another subtlety. The pattern is whenever ${PREPARE_SHELL} is used, we keep from using subprocess. Not sure why but it seems there is good reason.

Reply:
I think it is just that shell is used with inline scripts, but subprocess with script files.

Reply:
I think shell usage include both: 1) interactive (like create/rename/delete files, etc.); 2) scripting. Yeah, subprocess seems to focus on scripting (though it could also run ad-hoc binary command). But for simple shell interactive processing, it is still handy to use `shell.exec` task, like the following:
```
  "cleanup":
    - command: shell.exec
      params:
        script: |
          ${PREPARE_SHELL}
          rm -rf $DRIVERS_TOOLS || true
```

but at least theory, such usage could still be replaced by subprocess as follows:
```
      params:
        binary: bash
        args:
          - rm -rf $DRIVERS_TOOLS || true
```
right? If so, we might well eliminate the usage of `PREPARE_SHELL` completely.

Reply:
Resolving this (there are 3 comments that end up being about PREPARE_SHELL), and asking for either docs or a ticket in [this comment](https://github.com/mongodb/mongo-hibernate/pull/9#discussion_r1868288440).